pkg/iter/entry_iterator.go

```diff
@@ -404,8 +455,8 @@ func (i *entrySortIterator) Error() error {
 }
 
 func (i *entrySortIterator) Close() error {
-	for i.heap.Len() > 0 {
-		if err := i.heap.Pop().(EntryIterator).Close(); err != nil {
+	if len(i.is) > 0 {
+		if err := i.is[0].Close(); err != nil {
```

Comment:
That doesn't close all remaining iterator like before.

Reply:
I'm not sure I understand the original code then. It just closes the head of the heap doesn't it?

Reply:
Sorry. Misread `for` as an `if`.